src/utils/fs.ts

```diff
@@ -50,6 +50,34 @@ export function isFile (path: string): Promise<boolean> {
 }
 
 /**
+ * Verify a path exists and is a directory.
+ */
+export function isDirectory (path: string): Promise<boolean> {
+  return stat(path).then(stat => stat.isDirectory(), () => false)
+}
+
+function filterAsync<T>(items: T[], predicate: { (item: T): Promise<boolean> }): Promise<T[]> {
+  const pendingFilters = items.map(item => predicate(item))
+
+  return Promise.all(pendingFilters)
+    .then(filters => {
+      return items.filter((item: T, index: number) => filters[index])
+    })
+}
+
+/**
+ * Read the sub directories under in a path.
+ */
+export function readSubDirs (path: string): Promise<string[]> {
```

Comment:
This is a more thorough / robust solution because it will prune even if `main.d.ts` gets out of sync with the typings folder. This should never happen in practice because nobody should be messing with the typings folder, so I can really go either way on this.